packager/packager.cc

```diff
@@ -502,6 +509,41 @@ ast::ExpressionPtr prependPackageScope(const core::GlobalState &gs, ast::Express
     return scope;
 }
 
+uint8_t verifyLayer(core::MutableContext ctx, core::NameRef fun, ast::ExpressionPtr &expr) {
+    auto target = ast::cast_tree<ast::Literal>(expr);
+    const auto &packageDB = ctx.state.packageDB();
+    if ((target == nullptr && packageDB.getLayerCount() > 0) || !target->isString(ctx.state)) {
+        if (auto e = ctx.beginError(expr.loc(), core::errors::Packager::InvalidLayer)) {
+            string layers;
+            layers.append(packageDB.getLayerName(0));
+            for (int i = 1; i < packageDB.getLayerCount(); i++) {
+                layers.append(", ");
+                layers.append(packageDB.getLayerName(i));
+            }
+            e.setHeader("Argument to `{}` must be a valid layer name ({})", fun.show(ctx), layers);
+        }
+        return 0;
+    }
+
+    string layerName = target->asString(ctx.state).toString(ctx.state);
+    uint8_t layerRank = packageDB.getLayerRank(layerName);
+    if (layerRank == 255) {
+        if (auto e = ctx.beginError(expr.loc(), core::errors::Packager::InvalidLayer)) {
+            string layers;
+            layers.append(packageDB.getLayerName(0));
+            for (int i = 1; i < packageDB.getLayerCount(); i++) {
+                layers.append(", ");
+                layers.append(packageDB.getLayerName(i));
```

Comment:
It feels kind of weird here to use `getLayerName` instead of `layer.show(ctx)`

Reply:
I don't have a `Layer` here -- does `Layer(i).show(ctx)` seem more natural?

Reply:
I guess it depends on what you settle on in the PackageDB interface. If the signature is `vector<Layer> &layers()` you could so something like `fmt::map_join(packageDB.layers(), ", ", [ctx](auto layer) { return layer.show(ctx); })`